src/qibolab/transpilers/general_connectivity.py

```diff
@@ -0,0 +1,411 @@
+        i = 0
+        H.add_edge(self._circuit_repr[i][0], self._circuit_repr[i][1])
+        while GM.subgraph_is_monomorphic() == True:
+            result = GM
+            i = i + 1
+            H.add_edge(self._circuit_repr[i][0], self._circuit_repr[i][1])
+            GM = nx.algorithms.isomorphism.GraphMatcher(self._connectivity, H)
+            if self._connectivity.number_of_edges() == H.number_of_edges() or i == len(self._circuit_repr) - 1:
+                G = nx.relabel_nodes(self._connectivity, result.mapping)
+                self._graph = G
+                self._mapping = result.mapping
+                return
+        G = nx.relabel_nodes(self._connectivity, result.mapping)
+        self._graph = G
+        self._mapping = result.mapping
+
+    def greedy_init(self):
+        """initialize the circuit with greedy algorithm let a maximum number of 2-qubit
+        gates can be applied without introducing any SWAP gate"""
+        nodes = self._connectivity.number_of_nodes()
+        keys = list(self._connectivity.nodes())
+        values = list(range(nodes))
+        final_mapping = dict(zip(keys, values))
+        final_graph = nx.relabel_nodes(self._connectivity, final_mapping)
+        final_cost = len(self.reduce(final_graph))
+        for _ in range(self._init_samples):
+            random.shuffle(values)
+            mapping = dict(zip(keys, values))
+            graph = nx.relabel_nodes(self._connectivity, mapping)
+            cost = len(self.reduce(graph))
+            if cost == 0:
+                self._graph = graph
+                self._mapping = mapping
+                return
+            if cost < final_cost:
+                final_graph = graph
+                final_mapping = mapping
+                final_cost = cost
+        self._graph = final_graph
+        self._mapping = final_mapping
+
+    def init_qubit_map(self):
+        """Initial circuit-hardware qubit mapping.
+
+        Returns:
+            qubit_map (np.array): array containing the initial qubit mapping of the circuit.
+        """
+        qubit_map = np.zeros((len(self._mapping),), dtype=int)
+        for i, value in enumerate(self._mapping.values()):
+            qubit_map[value] = i
+        return qubit_map
+
+    def map_list(self, path):
+        """Return all possible walks of qubits for a given path.
+
+        Args:
+            path (list): path to move qubits.
+
+        Returns:
+            mapping_list (list): all possible walks of qubits for a given path.
+            meeting_point_list (list): all possible qubit meeting point in the path.
+        """
+        keys = path
+        path_ends = [path[0]] + [path[-1]]
+        path_middle = path[1:-1]
+        mapping_list = []
+        meeting_point_list = []
+        for i in range(len(path) - 1):
+            values = path_middle[:i] + path_ends + path_middle[i:]
+            mapping = dict(zip(keys, values))
```

Comment:
```suggestion
        path_ends = [path[0]] + [path[-1]]
        path_middle = path[1:-1]
        mapping_list = []
        meeting_point_list = []
        for i in range(len(path) - 1):
            values = path_middle[:i] + path_ends + path_middle[i:]
            mapping = dict(zip(path, values))
```

This has the double advantage of:
- avoiding an unneeded variable
- use a more telling name (more explicit), instead of the generic `keys`